HotelManagerment/com.spring.sample/src/com/spring/sample/controller/RegulationsController.java

```diff
@@ -1,22 +1,107 @@
 package com.spring.sample.controller;
 
+import java.util.List;
 import java.util.Locale;
 
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Controller;
 import org.springframework.ui.Model;
 import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.servlet.config.annotation.EnableWebMvc;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import com.spring.sample.entity.GuestTypes;
+import com.spring.sample.entity.RoomType;
+import com.spring.sample.entity.Setting;
+import com.spring.sample.entity.SurchargeRate;
+import com.spring.sample.service.RegulationsPageService;
 
 @Controller
-@EnableWebMvc
+@RequestMapping("/regulations")
 public class RegulationsController {
     private static final Logger logger = LoggerFactory.getLogger(RegulationsController.class);
 
-    @GetMapping(value = "/regulations")
+    @Autowired
+    private RegulationsPageService regulationsPageService;
+
+    @GetMapping
     public String index(Locale locale, Model model) {
         logger.info("Requesting regulation page");
+        List<RoomType> listOfRoomTypes = regulationsPageService.getAllRoomTypes();
+        List<GuestTypes> listOfGuestTypes = regulationsPageService.getAllGuestTypes();
+        List<Setting> listOfSettings = regulationsPageService.getAllSettings();
+        List<SurchargeRate> listOfSurchargeRates = regulationsPageService.getAllSurchargeRates();
+        model.addAttribute("roomTypes", listOfRoomTypes);
+        model.addAttribute("guestTypes", listOfGuestTypes);
+        model.addAttribute("settings", listOfSettings);
+        model.addAttribute("surchargeRates", listOfSurchargeRates);
         return "regulation/index";
     }
+
+    @PostMapping(value = "/update-room-type/{id}")
+    public String updateRT(@PathVariable Integer id, @RequestParam String typeRoomName,
+            @RequestParam Integer typeRoomPrice) {
+        regulationsPageService.updateRoomType(id, typeRoomName, typeRoomPrice);
+        return "redirect:/regulations";
+    }
+
+    @GetMapping(value = "/delete-room-type/{id}")
```

Comment:
method delete nha, ko phải get
```suggestion
    @DeleteMapping(value = "/id}")
```